src/helpers/rmq/RabbitMQ.ts

```diff
@@ -0,0 +1,285 @@
+import { Connection, connect, Options } from 'amqplib';
+import { setIntervalAsync, clearIntervalAsync, SetIntervalAsyncTimer } from 'set-interval-async/fixed';
+
+import loggerFactory from '../logger.js';
+import { RabbitMQChannel } from './RabbitMQChannel.js';
+import { RabbitMQPublish } from './handlers/RabbitMQPublish.js';
+import { RabbitMQConsumer } from './handlers/RabbitMQConsumer';
+
+const logger = loggerFactory('file:///rmq/RabbitMQ.ts');
+
+export type RabbitMQOptions = {
+  url: string;
+  reconnect: boolean;
+  reconnectTimeout: number;
+};
+
+const DEFAULT_OPTIONS: RabbitMQOptions = {
+  url: 'amqp://localhost:5672',
```

Comment:
Is this always the same url? if not, could this be added as a env variable?

Reply:
This is default options. This will be used if nothing was provided to constructor.